test/simple_api_reader-spec.ts

```diff
@@ -260,71 +307,162 @@ describe('simple_api_reader', () => {
-            ]
-        }), {});
+    describe('slicer', () => {
+        describe('will behave like elasticsearch date reader', () => {
+            const start = moment('2012-12-12T00:00:00.000Z');
+            const end = moment(start.toISOString()).add(1, 'minute');
+            let harness: SlicerTestHarness;
+            let updatedConfig: any;
 
-        beforeEach(async () => {
-            const query = {
+            async function makeSlicerTest(
+                config: any, numOfSlicers = 1,
+                recoveryData?: object[],
+                eventHook?: EventHook
+            ) {
+                const job = newTestJobConfig({
+                    analytics: true,
+                    slicers: numOfSlicers,
+                    operations: [
+                        config,
+                        {
+                            _op: 'noop'
+                        }
+                    ]
+                });
+                const myTest = new SlicerTestHarness(job, { assetDir, clients });
+                if (eventHook) myTest.events.on(eventHook.event, eventHook.fn);
+                await myTest.initialize(recoveryData);
+                return myTest;
+            }
+
+            const opConfig = {
+                _op: 'simple_api_reader',
+                date_field_name: 'created',
+                time_resolution: 's',
+                size: 100,
+                index: 'someindex',
+                interval: 'auto',
+                start: start.toISOString(),
+                end: end.toISOString(),
+                endpoint: baseUri,
                 token: 'test-token',
-                q: `created:[${start.toISOString()} TO ${end.toISOString()}} AND (slicer:query)`,
             };
 
-            scope.get(`/${testIndex}`)
-                .query(Object.assign({ size: 1 }, query))
-                .reply(200, {
-                    results: [{ created: start.toISOString() }],
-                    total: 1
-                });
+            async function waitForUpdate(config: any) {
+                const test = await makeSlicerTest(config, 1, [], { event: 'slicer:execution:update', fn: checkUpdate });
+                await pDelay(100);
+                return test;
+            }
 
-            scope.get(`/${testIndex}`)
-                .query(Object.assign({ size: 1 }, query))
-                .reply(200, {
-                    results: [{ created: end.toISOString() }],
-                    total: 1
-                });
 
-            scope.get(`/${testIndex}`)
-                .query(Object.assign({ size: 0 }, query))
-                .reply(200, {
-                    results: [],
-                    total: 2
-                });
+            function checkUpdate(updateObj: any) {
+                updatedConfig = get(updateObj, 'update[0]');
+                return true;
+            }
 
-            await harness.initialize([]);
+            beforeEach(async () => {
+                scope.get(/.*/)
```

Comment:
Why did you change this? Accepting any route can lead to tests passing but failing in real life.

```suggestion
                scope.get(`/${testIndex}`)
```

Reply:
Its not changed, this is just a preview error as the test was duplicated at that spot and the orignal one was moved, original test is kept as is with no changes

Reply:
Regardless it is still too generic, it should be specific to the index. I would caution you against being lazy in tests, be lazy in the implementation, not your tests.